### PR TITLE
Fix/requirements caching

### PR DESCRIPTION
**Pull request by ymaktepi**
This PR goes hand in hand with `feat/relax-requirements`.

Two changes:
the CI will install the latest versions of all the dependencies (via `pip-compile`) and run the tests

the CI also creates a `requirements-latest.txt`  file that will be used for caching the pip packages in order to speed up the testing process.

### CODE REVIEW

**Comment by tomasvanpottelbergh** [comment]
Looks good! I imagine Python 3.8 has been removed to speed things up? I would potentially leave it in the matrix to catch strange dependency conflicts. What do others think?

File: .github/workflows/merge.yml [hunk 1]
```diff
@@ -12,8 +12,8 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest]
-        python-version: [3.7, 3.8, 3.9]
-        flavour: ['core', 'all', 'pmdarima', 'torch', 'fbprophet']
+        python-version: [3.7, 3.9]
```
Comment (hrzn) [inline .github/workflows/merge.yml:15]:
```suggestion
        python-version: [3.7, 3.8, 3.9]
```

^ restoring 3.8 for merge workflow, as discussed

**Review by hrzn — APPROVED** [review]
LGTM! If I understand well, this does not include committing `requirements-latest.txt`, right? I guess it might be nice to have a followup later on to do this.

File: .github/workflows/merge.yml (hunk 1 above)
Comment (tomasvanpottelbergh) [reply .github/workflows/merge.yml:15]: And we could then only test 3.9 for PRs maybe?

File: .github/workflows/develop.yml [hunk 2]
```diff
@@ -11,8 +11,8 @@ jobs:
     strategy:
       matrix:
         os: [macos-latest, ubuntu-latest]
-        python-version: [3.7, 3.8, 3.9]
-        flavour: ['core', 'all']
+        python-version: [3.7, 3.9]
```
Comment (tomasvanpottelbergh) [inline .github/workflows/develop.yml:14]:
```suggestion
        python-version: [3.9]
```

**Review by tomasvanpottelbergh — APPROVED** [review]

**Pull request merged**